addons/sourcemod/scripting/tfgo/config.sp

```diff
@@ -30,12 +31,32 @@ public void ReadWeaponConfig(KeyValues kv)
 				int killAward = kv.GetNum("killAward", -1);
 				if (killAward <= -1)
 				{
-					char class[255];
+					char class[256];
 					TF2Econ_GetItemClassName(weapon.defindex, class, sizeof(class));
 					g_weaponClassKillAwards.GetValue(class, killAward);
 				}
 				weapon.killAward = killAward;
 				
+				// Variants
+				char variantsString[256];
+				kv.GetString("variants", variantsString, sizeof(variantsString));
+				ArrayList variantList = new ArrayList();
+				if (strlen(variantsString) > 0)
+				{
+					char buffers[32][256]; // max. 32 variants
+					for (int i = 0; i < sizeof(buffers); i++)buffers[i] = "-1"; // because 0 = Bat
+					ExplodeString(variantsString, ";", buffers, sizeof(buffers), sizeof(buffers[]));
+					
+					for (int i = 0; i < sizeof(buffers); i++)
```

Comment:
`ExplodeString` returns number of strings retrieved. so you can replace loop `sizeof(buffers)` to count from `ExplodeString`, without needing to set all `-1` first.
https://sm.alliedmods.net/new-api/string/ExplodeString